janitor/functions.py

```diff
@@ -4055,6 +4264,89 @@ def take_a_if_even(x):
     return df
 
 
+@pf.register_dataframe_method
+def join_conditional(
```

Comment:
@hectormz could I assign this one to you to review please?

Reply:
Great!

Reply:
You should be able to use it on the same dataframe, like a self join. However, this does not match the expected output, as every value in one column should be compared to every other value in the other column. This is achievable with a cartesian product, which is memory intensive at least. So, for the example used in the test, the output should be : 

```python
  SalesMonth  Company1  Company2
0        Jan     150.0     180.0
1        Jan     150.0     250.0
2        Jan     150.0     500.0
3        Feb     200.0     250.0
4        Feb     200.0     500.0
5        Mar     300.0     500.0
6      April     400.0     500.0
```

where every value in `company1` column is compared against every value in `company2` columnn, and only the values where `company1` is less than `company2` is returned.

Reply:
That's a good point, @samukweku. @BaritoneBeard, please see the follow-up thread below for the action item.